### PR TITLE
Add gevent example

### PR DESCRIPTION
**Pull request by lewoudar**
update some files
README.rst: update the snippet so that it is up to date with the current
release
.gitignore: add .idea to ignore pycharm settings folder
examples: add gevent example in its own folder
docs/source/: create a gevent-example.rst
docs/source/examples.rst: add a link to gevent-example.rst
CONTRIBUTORS.rst: add a reference to my name and work

### CODE REVIEW

**Comment by lewoudar** [comment]
Hello,
could someone explain to me how to avoid the error that occurs on python2.7
At the same time I wonder if it is still worth maintaining this version since it only has three months left.

File: .gitignore [hunk 1]
```diff
@@ -9,3 +9,4 @@ __pycache__
 .hypothesis/
 .cache/
 _trial_temp/
+.idea/
```
Comment (Kriechi) [inline .gitignore:12]: IDE / editor-specific ignores are best placed in your personal (device/OS-specific) global gitignore.

File: examples/gevent/gevent-server.py [hunk 2]
```diff
@@ -0,0 +1,187 @@
+# -*- coding: utf-8 -*-
+"""
+gevent-server.py
+================
+
+A simple HTTP/2 server written for gevent serving static files from a directory specified as input.
+If no directory is provided, the current directory will be used.
+
+Requires Python 3.6+.
```
Comment (Kriechi) [inline examples/gevent/gevent-server.py:9]: FYI: hyper-h2 (with current git-master) now requires at least py36.

**Comment by Kriechi** [comment]
Hi! Thanks for sending this new example. Would you mind rebasing to the current master and giving it another look at and manual test. There have been some changes - just want to make sure your example still works the same as you intended.
We recently deprecated a bunch of stuff and Python versions - so I hope this makes it easier!

File: examples/gevent/gevent-server.py (hunk 2 above)
Comment (lewoudar) [reply examples/gevent/gevent-server.py]: I remove the requirement in the docstring, thanks!

File: .gitignore (hunk 1 above)
Comment (lewoudar) [reply .gitignore]: ok

**Comment by lewoudar** [comment]
Thanks @Kriechi for reviewing my pull request, I was feeling a bit lonely. Normally I fixed all the issues and the example works as expected

**Comment by Kriechi** [comment]
Thanks! Almost perfect now!
Did you change your email or git author data? GitHub seems to not recognize you anymore on the individual commits... do you want to fix this - or should I merge as-is?

**Comment by lewoudar** [comment]
Yeah I use a different email in my commit messages than what I use to login. It is not a real issue for me, you can merge it as it.

**Comment by Kriechi** [comment]
Wohoo! Thanks for the new example! 🎉 👍

**Comment by lewoudar** [comment]
Yeah !! 🤟🏾

**Pull request merged**